bcal.c

```diff
@@ -621,6 +622,263 @@ bool lba2chs(char *lba, t_chs *p_chs)
 	return TRUE;
 }
 
+void InputError()
+{
+	fprintf(stderr, "\nPlease enter inputs properly. Always put spece between each two arguments.\n\
+And a byte unit can only be divided or multiplied by a plain integer.\nExample:\nbcal \"( 5kb + 2mb ) / 3\"\nbcal \"5tb / 12\"\nbcal \"2.5mb * 3\" etc.\n\n");
+	exit(-1);
+}
+
+/* Convert any unit in bytes */
+maxuint_t unitconv(Data bunit, short *u)
+{
+		char *numstr = bunit.p;
+
+		if(numstr == NULL) {
+			InputError();		
+		}
+
+		char *num;
+		char unit[4] = {'\0'};
+		int len = strlen(numstr);
+
+		if (isalpha(numstr[len - 3])) {
```

Comment:
I did not understand, there is no space between number and unit. I am taking strings like "25kb","3mb" etc. Otherwise it will make problem with strtok.
I will work on using strstr.